src/BaseComponent.php

```diff
@@ -87,6 +85,13 @@ protected function loadConfig(): void
         $this->logger->debug('Config loaded');
     }
 
+    protected function getFormattedConfig(): array
```

Comment:
getRawConfigData ?

Reply:
To neni uplne spatne, ale muze to vzbuzovat pocit, ze to bude jen vytazeny string ze souboru tj. ze je to stale json.

Reply:
Takhle to vzbuzuje pocit, že mi to vrátí pretty printed json IMHO :) 

Tak getDeserializedConfigData?

Reply:
Naming things = #hardProblem

Reply:
Asi bych to nechal na to Raw - je to takove nejjednoznacnejsi a pres ten prvni comment. Return type `array` by mel pomoct trknout, ze to neni json.